spectator/registry.go

```diff
@@ -0,0 +1,214 @@
+// Package spectator provides a minimal Go implementation of the Netflix Java
+// Spectator library. The goal of this package is to allow Go programs to emit
+// metrics to Atlas.
+//
+// Please refer to the Java Spectator documentation for information on
+// spectator / Atlas fundamentals: https://netflix.github.io/spectator/en/latest/
+package spectator
+
+import (
+	"encoding/json"
+	"github.com/Netflix/spectator-go/spectator/meter"
+	"github.com/Netflix/spectator-go/spectator/writer"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// Config represents the Registry's configuration.
+type Config struct {
+	CommonTags map[string]string `json:"common_tags"`
+	Log        Logger
+}
+
+// Meter represents the functionality presented by the individual meter types.
+type Meter interface {
+	MeterId() *meter.Id
+}
+
+// RegistryInterface was extracted from Registry. It's used to make sure we're honoring the same API for thin and fat client, might delete later.
+type RegistryInterface interface {
+	GetLogger() Logger
+	SetLogger(logger Logger)
+	NewMeter(id *meter.Id, meterFactory MeterFactoryFun) Meter
+	NewId(name string, tags map[string]string) *meter.Id
+	Counter(name string, tags map[string]string) *meter.Counter
+	CounterWithId(id *meter.Id) *meter.Counter
+	Timer(name string, tags map[string]string) *meter.Timer
+	TimerWithId(id *meter.Id) *meter.Timer
+	Gauge(name string, tags map[string]string) *meter.Gauge
+	GaugeWithId(id *meter.Id) *meter.Gauge
+	MaxGauge(name string, tags map[string]string) *meter.MaxGauge
+	MaxGaugeWithId(id *meter.Id) *meter.MaxGauge
+	DistributionSummary(name string, tags map[string]string) *meter.DistributionSummary
+	DistributionSummaryWithId(id *meter.Id) *meter.DistributionSummary
+	PercentileDistributionSummary(name string, tags map[string]string) *meter.PercentileDistributionSummary
+	PercentileDistributionSummaryWithId(id *meter.Id) *meter.PercentileDistributionSummary
+	PercentileTimer(name string, tags map[string]string) *meter.PercentileTimer
+	PercentileTimerWithId(id *meter.Id) *meter.PercentileTimer
```

Comment:
Not sure why the Percentile* meters where not part of the Registry API. Instead you'd create them by passing a Registry. 
I added methods here for consistency with other Meters

Reply:
In the original spectator-go implementation? Not sure.

Best to keep them all together.

Reply:
Makes sense 👍